### PR TITLE
microsite-next: Add Sidebar, make references compatible

### PR DESCRIPTION
**Pull request by jhaals**
- Add current sidebar to the new microsite.
- Fixes reference docs compatibility by stripping out html comments which isn't allowed in docu 2.x.
- Adds placeholder /docs/reference page if api docs is not generated since docu 2.x require them to be available at build time.

### CODE REVIEW

File: microsite-next/scripts/pre-build.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+description: "Index of all Backstage Packages"
+---
+Run \`yarn build:api-docs\` to generate the API docs.
+`;
+
+async function main() {
+  const referencesDir = '../docs/reference';
+  if (existsSync(referencesDir)) {
+    console.log('Removing HTML comments from docs/reference folder');
+    await replace({
```
Comment (marcuseide) [inline microsite-next/scripts/pre-build.js:33]: Just a nit but it seems that `replace()` is not `async`

File: microsite-next/scripts/pre-build.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      regex: '<!--(.*?)-->',
```
Comment (marcuseide) [inline microsite-next/scripts/pre-build.js:34]: Do we need to support multiline comments? 😅

File: microsite-next/scripts/pre-build.js (hunk 1 above)
Comment (jhaals) [reply microsite-next/scripts/pre-build.js]: I did not notice that, thanks 👍

File: microsite-next/scripts/pre-build.js (hunk 2 above)
Comment (jhaals) [reply microsite-next/scripts/pre-build.js:34]: I don't think so, at least not for now as no builds have exploded because of the references so far 😅

**Review by marcuseide — APPROVED** [review]

File: microsite-next/package.json [hunk 3]
```diff
@@ -4,9 +4,8 @@
   "license": "Apache-2.0",
   "private": true,
   "scripts": {
-    "examples": "docusaurus-examples",
-    "start": "docusaurus start",
-    "build": "docusaurus build",
+    "start": "node scripts/pre-build.js && docusaurus start",
+    "build": "node scripts/pre-build.js && docusaurus build",
```
Comment (Rugvip) [inline microsite-next/package.json:8]: 👌

File: microsite-next/sidebars.json [hunk 4]
```diff
@@ -1 +1,366 @@
-{}
+{
+  "releases": {
+    "Release Notes": [
+      "releases/v1.8.0",
```
Comment (Rugvip) [inline microsite-next/sidebars.json:4]: I think we can live with this being slightly out of sync 😅

**Review by Rugvip — APPROVED** [review]

File: microsite-next/sidebars.json (hunk 4 above)
Comment (jhaals) [reply microsite-next/sidebars.json:4]: Yeah it's good enough for now :)

**Pull request merged**